go/cmd/dolt/commands/show.go

```diff
@@ -0,0 +1,265 @@
+	ap.SupportsInt(limitParam, "", "record_count", "limits to the first N diffs.")
+	ap.SupportsFlag(cli.CachedFlag, "c", "Show only the staged data changes.")
+	ap.SupportsFlag(SkinnyFlag, "sk", "Shows only primary key columns and any columns with data changes.")
+	ap.SupportsFlag(MergeBase, "", "Uses merge base of the first commit and second commit (or HEAD if not supplied) as the first commit")
+	ap.SupportsString(DiffMode, "", "diff mode", "Determines how to display modified rows with tabular output. Valid values are row, line, in-place, context. Defaults to context.")
+	return ap
+}
+
+// Exec executes the command
+func (cmd ShowCmd) Exec(ctx context.Context, commandStr string, args []string, dEnv *env.DoltEnv) int {
+	ap := cmd.ArgParser()
+	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, showDocs, ap))
+	apr := cli.ParseArgsOrDie(ap, args, help)
+
+	opts, err := parseShowArgs(ctx, dEnv, apr)
+	if err != nil {
+		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+	}
+
+	opts.diffDisplaySettings = parseDiffDisplaySettings(ctx, dEnv, apr)
+
+	err = showCommits(ctx, dEnv, opts)
+
+	return handleErrAndExit(err)
+}
+
+func (cmd ShowCmd) validateArgs(apr *argparser.ArgParseResults) errhand.VerboseError {
+	if apr.Contains(StatFlag) || apr.Contains(SummaryFlag) {
+		if apr.Contains(SchemaFlag) || apr.Contains(DataFlag) {
+			return errhand.BuildDError("invalid Arguments: --stat and --summary cannot be combined with --schema or --data").Build()
+		}
+	}
+
+	f, _ := apr.GetValue(FormatFlag)
+	switch strings.ToLower(f) {
+	case "tabular", "sql", "json", "":
+	default:
+		return errhand.BuildDError("invalid output format: %s", f).Build()
+	}
+
+	return nil
+}
+
+func parseShowArgs(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgParseResults) (*showOpts, error) {
+
+	decorateOption := apr.GetValueOrDefault(cli.DecorateFlag, "auto")
+	switch decorateOption {
+	case "short", "full", "auto", "no":
+	default:
+		return nil, fmt.Errorf("fatal: invalid --decorate option: %s", decorateOption)
+	}
+
+	return &showOpts{
+		showParents: apr.Contains(cli.ParentsFlag),
+		decoration:  decorateOption,
+		specRefs:    apr.Args,
+	}, nil
+}
+
+func showCommits(ctx context.Context, dEnv *env.DoltEnv, opts *showOpts) error {
+	if len(opts.specRefs) == 0 {
+		return showCommit(ctx, dEnv, opts, dEnv.RepoStateReader().CWBHeadSpec())
+	}
+
+	for _, specRef := range opts.specRefs {
+		commitSpec, err := getCommitSpec(specRef)
+
+		if err != nil {
+			cli.PrintErrln(color.HiRedString("Fatal error: invalid commit spec %s", specRef))
+			return err
+		}
+
+		err = showCommit(ctx, dEnv, opts, commitSpec)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func showCommit(ctx context.Context, dEnv *env.DoltEnv, opts *showOpts, commitSpec *doltdb.CommitSpec) error {
+
+	comm, err := dEnv.DoltDB.Resolve(ctx, commitSpec, dEnv.RepoStateReader().CWBHeadRef())
+
```

Comment:
(minor/nit) Could you remove the newlines between the statement returning an `err` and the `err` checking if block? (lines 168, 175, and 198) Most of the codebase follows that style and it just makes the error handling feel a little more connected to the code that generated the error.